android/java/org/chromium/chrome/browser/CrossPromotionalModalDialogFragment.java

```diff
@@ -0,0 +1,109 @@
+        super.onConfigurationChanged(newConfig);
+        setDialogParams();
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        getDialog().setOnKeyListener(new DialogInterface.OnKeyListener() {
+            @Override
+            public boolean onKey(android.content.DialogInterface dialog,
+                                 int keyCode, android.view.KeyEvent event) {
+                if ((keyCode == android.view.KeyEvent.KEYCODE_BACK)) {
+                    dismiss();
+                    return true;
+                } else return false;
+            }
+        });
+        setDialogParams();
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.fragment_cross_promotional_modal_dialog, container, false);
+    }
+
+    @Override
+    public void onViewCreated(View view, Bundle savedInstanceState) {
+        Button mDoneButton = view.findViewById(R.id.btn_done);
+        mDoneButton.setOnClickListener(this);
+
+        String mPromoText = getResources().getString(R.string.download_brave);
+        int indexOfSpan = mPromoText.indexOf(getResources().getString(R.string.brave_url));
+
+        TextView mCrossPromoText = view.findViewById(R.id.cross_modal_text);
+
+        Spanned textToSpan = BraveRewardsHelper.spannedFromHtmlString(mPromoText);
+        SpannableString ss = new SpannableString(textToSpan.toString());
+
+        ForegroundColorSpan foregroundSpan =
+            new ForegroundColorSpan(getResources().getColor(R.color.shield_back_button_tint));
+        ss.setSpan(foregroundSpan, indexOfSpan,
+                   (indexOfSpan + getResources().getString(R.string.brave_url).length()), Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
+        mCrossPromoText.setText(ss);
+
+    }
+
+    @Override
+    public void onClick(View view) {
+        dismiss();
+    }
+
+    private void setDialogParams() {
+        DisplayMetrics displayMetrics = new DisplayMetrics();
+        getActivity().getWindowManager().getDefaultDisplay().getMetrics(displayMetrics);
+        int mDeviceHeight = displayMetrics.heightPixels;
+        int mDeviceWidth = displayMetrics.widthPixels;
+
+        WindowManager.LayoutParams params = getDialog().getWindow().getAttributes();
+        boolean isTablet = DeviceFormFactor.isNonMultiDisplayContextOnTablet(getActivity());
+        boolean isLandscape = ConfigurationUtils.isLandscape(getActivity());
+        if (isTablet) {
+            params.width = (int) (0.5 * mDeviceWidth);
+        } else {
+            if (isLandscape) {
+                params.width = (int) (0.5 * mDeviceWidth);
+            } else {
+                params.width = (int) (0.9 * mDeviceWidth);
+            }
+        }
+        params.height = LinearLayout.LayoutParams.WRAP_CONTENT;
+        getDialog().getWindow().setAttributes(params);
+    }
+}
```

Comment:
nit: i chromium they add empty line in java files, I think we should it too for consistency